# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraphImpl.h
# Commit: 60a6baeddbff
# Full Hash: 60a6baeddbff089de8a6e00b31643c73469bf2ce
# Author: Paul Adenot <paul@paul.cx>
# Date: 2020-01-24 21:46:30
# Regressor Bug: 1608505
# File Overlap Count: 1
# Description:
#   Bug 1608505 - Cap the audio output channel count to something the device can handle. r=achronop
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60549
# ==============================================================================

diff -r 2a0d5d66eb16 -r 60a6baeddbff dom/media/MediaTrackGraphImpl.h
--- a/dom/media/MediaTrackGraphImpl.h	Fri Jan 24 08:55:51 2020 +0000
+++ b/dom/media/MediaTrackGraphImpl.h	Fri Jan 24 11:23:37 2020 +0000
@@ -485,7 +485,10 @@
     mTrackOrderDirty = true;
   }
 
+  // Get the current maximum channel count. Graph thread only.
   uint32_t AudioOutputChannelCount() const;
+  // Set a new maximum channel count. Graph thread only.
+  void SetMaxOutputChannelCount(uint32_t aMaxChannelCount);
 
   double AudioOutputLatency();
 
@@ -997,6 +1000,14 @@
    * whenever the audio device running this MediaTrackGraph changes.
    */
   double mAudioOutputLatency;
+
+  /**
+   * The max audio output channel count the default audio output device
+   * supports. This is cached here because it can be expensive to query. The
+   * cache is invalidated when the device is changed. This is initialized in the
+   * ctor, and the read/write only on the graph thread.
+   */
+  uint32_t mMaxOutputChannelCount;
 };
 
 }  // namespace mozilla
